### DISCUSSION TITLE
How to create a node exactly where mouse pointer is ?

### DISCUSSION THREAD

**Discussion by gitHoussem** [bcakmakoglu/vue-flow#1320] [Q&A] [answered]
Hello,

I'm struggling to create a node where my mouse pointer is inside the vueflow viewport.

The mouse positions I get from the onPaneClick are not the same as the positions of the nodes inside the viewport.

I found that the created of reactflow answered this question here https://github.com/xyflow/xyflow/discussions/3112

I am wondering if vueflow has something equivalent to it ?

Thanks

**Comment by bcakmakoglu** [comment]
Check out the DnD example: https://vueflow.dev/examples/dnd.html

That should make it clear how to add a node at mouse position (using `screenToFlowCoordinate`)

**Reply by bcakmakoglu** [reply, marked as answer]
TL;DR:

```ts
const { screenToFlowCoordinate } = useVueFlow()

function onSomeEvent(event) {
	const position = screenToFlowCoordinate({
		x: event.clientX,
		y: event.clientY,
	})
}

**Reply by gitHoussem** [reply]
Thanks man ! This is exactly it

**Reply by AkariMarisa** [reply]
I have a dump question. How can I achieve the same result using @vue-flow/core@1.23.0? There's no screenToFlowCoordinate function in useVueFlow.
